tests/test_base_device.py

```diff
@@ -994,75 +902,20 @@ def test_estimator_with_various_multi_qubit_pauli_obs(
         sampler_execute.assert_not_called()
         estimator_execute.assert_called_once()
 
-        assert np.allclose(res, expectation, atol=0.3)  ## atol is high due to high variance
-
-    @pytest.mark.parametrize(
-        "measurements, expectation",
-        [
-            ([qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliX(0))], (1, 0)),
-            ([qml.var(qml.PauliX(0))], (1)),
-            (
-                [
-                    qml.expval(qml.PauliX(0)),
-                    qml.expval(qml.PauliZ(0)),
-                    qml.var(qml.PauliX(0)),
-                ],
-                (0, 1, 1),
-            ),
-            ([qml.expval(0.5 * qml.Y(0) + 0.5 * qml.Y(0) - 1.5 * qml.X(0) - 0.5 * qml.Y(0))], (0)),
-            (
-                [
-                    qml.expval(
-                        qml.ops.LinearCombination(
-                            [1, 3, 4], [qml.X(3) @ qml.Y(2), qml.Y(4) - qml.X(2), qml.Z(2) * 3]
-                        )
-                        + qml.X(4)
-                    )
-                ],
-                (16),
-            ),
-        ],
-    )
-    def test_process_estimator_job(self, measurements, expectation):
-        """Tests that the estimator returns expected and accurate results for an ``expval`` and ``var`` for a variety of multi-qubit observables"""
-
-        # make PennyLane circuit
-        qs = QuantumScript([], measurements=measurements)
-
-        # convert to Qiskit circuit information
-        qcirc = circuit_to_qiskit(qs, register_size=qs.num_wires, diagonalize=False, measure=False)
-        pauli_observables = [mp_to_pauli(mp, qs.num_wires) for mp in qs.measurements]
-
-        # run on simulator via Estimator
-        estimator = Estimator(backend=backend)
-        result = estimator.run([qcirc] * len(pauli_observables), pauli_observables).result()
-
-        # confirm that the result is as expected - if the test fails at this point, its because the
-        # Qiskit result format has changed
-        assert isinstance(result, EstimatorResult)
-
-        assert isinstance(result.values, np.ndarray)
-        assert result.values.size == len(qs.measurements)
-
-        assert isinstance(result.metadata, list)
-        assert len(result.metadata) == len(qs.measurements)
-
-        for data in result.metadata:
-            assert isinstance(data, dict)
-        processed_result = QiskitDevice2._process_estimator_job(qs.measurements, result)
-        assert isinstance(processed_result, tuple)
-        assert np.allclose(processed_result, expectation, atol=0.1)
+        assert np.allclose(res[0], expectation, atol=0.3)  ## atol is high due to high variance
 
     @pytest.mark.parametrize("num_wires", [1, 3, 5])
     @pytest.mark.parametrize("num_shots", [50, 100])
-    @pytest.mark.skip(reason="Need to replace this with using SamplerV2.")
+    @pytest.mark.skip(
+        reason="Need to replace this with using SamplerV2."
+    )  # ToDo: Do we even need this?
```

Comment:
should this still be being skipped? Shouldn't it work now? We do definitely want a unit test for `generate_samples`, so unless this is duplicated elsewhere we should keep it.

Reply:
This does not work right now cause we can't set a shots value yet (no processing kwargs). I would like to keep a test for `generate_samples` but it is actually kind of tricky to update this function to use the new `_execute_sampler` function.

Reply:
Well, I thought it was tricky but it's implemented in the #547 now.